src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs

```diff
@@ -87,6 +88,7 @@ public virtual IDisposable Start()
                 _applicationLifetime.NotifyStopping();
                 server.Dispose();
                 _applicationLifetime.NotifyStopped();
+                loggerFactory.Dispose();
```

Comment:
@HaoK  and I discussed about this and agreed that disposing loggerfactory alone for this PR is fine and we will file another issue to handle disposing all the hosting engine services as it requires more refactoring than expected.

@Tratcher